contributors/design-proposals/node/cache-management.md

```diff
@@ -0,0 +1,220 @@
+establish cache as a first class resource, that is a resource that is
+explicitly requested at container launch time. But seeking to control cache
+allocation to a container based on it QoS.
+
+## Proposed Changes
+
+### Extending CPU Manager
+
+The original [cpu manager design proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/cpu-manager.md) alluded to extending its scope
+to control processor cache allocation to obtain more predictable performance. 
+The CPU Manager intercepts all container related actions, in particular add,
+remove, start, and state update. Below we discuss changes for cache allocation. 
+
+### Pod QoS Class
+
+Kubernetes supports a notion of [pod QoS](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/), the higher the value, the higher
+the priority. We treat the Guaranteed QOS pod as deserving of more cache
+resources.
+
+### Configuring CPU Manager Cache Resource
+
+It further comprehends three policies, namely “none”, “static”, and “dynamic”,
+and policy is selected through kubelet configuration. 
+
+ - **None** The cache resource allocation feature is still experimental, and
+   thus the default policy will stay “none” and have  no effect on cache
+   reservation. 
+ - **Static** While for CPUs the Static policy has logical CPU pools, in
+   particular “SHARED”, “RESERVED”, ASSIGNABLE, and “EXCLUSIVE ALLOCATIONS”,
+   we shall define for Cache just two COS, namely **“SHARED” and “RESERVED”**.
+   The shared pool could be half the available cache ways or any other
+   fraction, alternately we could define multiple cache pools with different
+   degrees of overlap.  Without loss of generality, for the remainder of this
+   discussion, let us consider just two classes of service, namely SHARED and
+   RESERVED. Given the fixed assignments, we have the option as a first pass to
+   not modify the “state update” code. Alternately, we could modify the update
+   code, possibly in an out of band manner, to indicate the number of
+   containers assigned to the configured COS. This would be useful in
+   implementing back pressure, that is, to decline  launch requests for high
+   QOS containers on a heavily subscribed high QOS COS host.
+ - **Dynamic**  True dynamic policy will not be implemented at this time. 
+
+### Alternative Cache Configuration
+
+Instead of just two (or more) overlapping COS, one might ask why not partition
+the cache and assign each exclusively to a CPU core, along the lines of CPU
+pinning. The underlying cache management library provides such support, but in
+this experimental  release, we opt for overlapping pools because of the greater
+sharing they provide (leading to better best-effort performance) yet providing
+less contention for workloads that have a higher QOS requirement. Further, when
+workloads are pinned to a CPU core, they already have exclusive access to the
+core’s L1 and L2 cache.
```

Comment:
How does hyperthreading impact this assumption?

Reply:
Thanks for pointing out this, @vishh. Absolutely two logical thread will share the L1 and L2 cache in one cpu core. Will re-work this part.